### ISSUE TITLE
src/SimpPDB.cc: 42: ERROR: non TER line starting fragment in fragments_DB

### ISSUE THREAD

**Issue by sarah-robinson** [UnixJunkie/fragger#5]
Hello,

I followed the instructions in the readme and when I get to the command to query with a fragment, I get the following error message, and the output results file is empty. 

`src/SimpPDB.cc: 42: ERROR: non TER line starting fragment in:      27.112 -65.799   8.836  1.00 56.70           C
`

I've attached all the files I've used. This coordinate is not in my test fragment, but I can see it is in the pdb structure that I have taken the test fragment from. I've looked in the fragment_DB and the coordinate is in the middle of the structure.  I've repeated this with different fragments from different pdb structures and get the same error message, with different coordinates - it seems to consistently be for a Calpha atom and is contained within the structure that the fragment comes from. 

Thank you so much for your help - and please let me know what further information I can provide.

kerin:fragger sarah$ ./src/query.native -db fragments_DB -q 6e9i_A_0001.pdb -d 0.2 -o results -idx 6e9i_A_0001.rmsds
2020-05-15 12:01:01.123 INFO : loading DB index from previous run ...
2020-05-15 12:01:01.128 INFO : loaded
2020-05-15 12:01:01.331 INFO : index: 6e9i_A_0001.rmsds
running:
/Users/sarah/fragger/ext/ranker_AA_QCP/ranker_aa 9 /var/folders/4v/549rj1s12r18pqn9rc5gbwdh0000gs/T/ref_.tmp.c14d19.pdb 6e9i_A_0001.pdb 0
2020-05-15 12:01:01.412 INFO : loading index from previous run ...
2020-05-15 12:01:01.419 INFO : loaded
2020-05-15 12:01:01.419 INFO : first: 0.000001
2020-05-15 12:01:01.420 INFO : last: 0.000001
2020-05-15 12:01:01.512 INFO : nb candidates: 1
2020-05-15 12:01:01.562 INFO : loading 1 fragments ...
2020-05-15 12:01:01.562 INFO : filtering them ...
running:
/Users/sarah/fragger/ext/ranker_AA_QCP/ranker_aa 9 results 6e9i_A_0001.pdb 0.200000 -bin
src/qcprot.cc:261:FastCalcRMSD: did not converge for 0 0
src/SimpPDB.cc: 42: ERROR: non TER line starting fragment in:      27.112 -65.799   8.836  1.00 56.70           C  
2020-05-15 12:01:01.633 INFO : writing them ...
2020-05-15 12:01:01.633 INFO : Found 0 fragment(s)
2020-05-15 12:01:01.633 INFO : Done
kerin:fragger sarah$ 

[fragger.zip](https://github.com/UnixJunkie/fragger/files/4633931/fragger.zip)

**Comment by UnixJunkie** [comment]
Hi Sara,

Have you read and followed the instructions in there?
https://github.com/UnixJunkie/fragger/blob/master/README

If you managed to install the software, the rest should be a piece of cake.

**Comment by sarah-robinson** [comment]
Yes, if I run the instructions and commands exactly as they are in README (literally copying and pasting the full README, having already sourced ccp4), I still get this same error - full output attached. The program still appears to complete, but without any fragments identified and nothing written in results.  



[fragger_TER_error.txt](https://github.com/UnixJunkie/fragger/files/4640022/fragger_TER_error.txt)

**Comment by UnixJunkie** [comment]
I will try to have a look today.

**Comment by UnixJunkie** [comment]
Some commands from the README file are not OK to copy-paste.
```
./src/query.native -db fragments_DB -i 01_A_0001:fragments_9 -o 01_A_0001.rmsds
```
replace 01_A_0001 by 6e9i_A_0001, for example.
I.e. a valid fragment id for you set of PDB files.

**Comment by UnixJunkie** [comment]
In this one:
```
./src/query.native -db fragments_DB -i 06_A_0049:fragments_9 -o 06_A_0049.rmsds
```
replace 06_A_0049 by 6e9u_B_0207 (for example).

**Comment by UnixJunkie** [comment]
Maybe this will help you:
```
~/src/fragger/ext/ranker_AA_QCP/ranker_aa 9 fragments_9 6e9i_A_0001.pdb 0.2
0.000001 6e9i_A_0001
```
This is a brute force query against all the fragments_9 against your query fragment
(6e9i_A_0001).
It shows that there is a single fragment nearer than 0.2 RMSD (and it is itself).

**Comment by sarah-robinson** [comment]
Sorry, I should have been clearer. If I run the instructions exactly as they are in the README, I run it on the data that is included in the Fragger GitHub /data/ folder so fragment ID 01_A_0001 is valid. When I do this, I still get the same "src/SimpPDB.cc: 42: ERROR: non TER line starting fragment in fragments_DB" error message as I do when I run it on my own test set (where I run it with appropriately revised commands e.g. changing the fragment names).

**Comment by UnixJunkie** [comment]
There is a Fragger user which might help you soon.
He has already seen this error.
He is busy with a deadline until tomorrow.
I an personally busy with a deadline until the end of the month.

**Comment by UnixJunkie** [comment]
Some of your PDBs have problems: 4k3d.pdb: some residues have several CAs.

**Comment by UnixJunkie** [comment]
This is what you have
```
ATOM     33  N  AARG H   4       8.179  -7.988  22.122  0.64 38.90           N  
ATOM     34  N  BARG H   4       8.146  -8.030  22.154  0.36 38.52           N  
ATOM     35  CA AARG H   4       8.476  -9.031  21.142  0.64 40.04           C  
ATOM     36  CA BARG H   4       8.465  -9.024  21.134  0.36 39.64           C  
ATOM     37  C  AARG H   4       9.063 -10.236  21.840  0.64 37.07           C  
ATOM     38  C  BARG H   4       9.095 -10.255  21.760  0.36 36.97           C  
ATOM     39  O  AARG H   4      10.153 -10.159  22.403  0.64 36.90           O  
ATOM     40  O  BARG H   4      10.241 -10.215  22.201  0.36 37.37           O  
```
What we expect is more like this:
```
ATOM     16  N   GLN H   2       6.601  -1.569  22.977  1.00 38.05           N  
ATOM     17  CA  GLN H   2       6.667  -2.637  21.985  1.00 39.66           C  
ATOM     18  C   GLN H   2       7.256  -3.883  22.626  1.00 39.44           C  
ATOM     19  O   GLN H   2       8.374  -3.859  23.141  1.00 38.79           O  
```

**Comment by UnixJunkie** [comment]
I wonder if those are not due to residues with alternate positions.
You should clean your PDBs further before feeding them to fragger.
Our pdb cleaning script cannot handle those.

**Comment by UnixJunkie** [comment]
A backbone atom should never be repeated.
There should be a strict alternance of N,CA,C,O.

**Comment by UnixJunkie** [comment]
Chimera can fix those as part of the docking preparation procedure:
https://www.cgl.ucsf.edu/chimera/docs/ContributedSoftware/dockprep/dockprep.html
Look for 'alternate locations' in there.
Maybe you should ask a crystallographer or whoever gave you those PDB
to do this for you.
The problem is that this chimera script might do other changes to your PDBs, that you don't want.

**Comment by sarah-robinson** [comment]
Thank you very much for your help. Does this also explain why I can't get Fragger to run, even on the data that is provided in the Fragger /data/ folder? (10 PDB files, labelled 00-09)

**Comment by UnixJunkie** [comment]
No idea.
Try with a single of your PDB files and inspect what is produced in the cleaned/ directory.
Also, deal with your 'alternate locations' first.
